virtio-devices/src/block.rs

```diff
@@ -616,7 +616,9 @@ impl Block {
                 (disk_nsectors, avail_features, 0, config, false)
             };
 
-        let serial = serial.map(Vec::from).unwrap_or(build_serial(&disk_path));
+        let serial = serial
+            .map(Vec::from)
+            .unwrap_or_else(|| build_serial(&disk_path));
```

Comment:
Also, why these changes are needed? If they are for accommodating clippy warnings, please put it in a separate commit.

Reply:
Done.

Reply:
It's very strange that this wasn't found by the stable/beta/nightly runs... :confused: